src/expandableCalendar/agendaListNew.tsx

```diff
@@ -0,0 +1,259 @@
+      } else {
+        sectionTitle = new XDate(title).toString(dayFormat);
+      }
+    }
+
+    if (markToday) {
+      const string = getDefaultLocale().today || todayString;
+      const today = isToday(title);
+      sectionTitle = today ? `${string}, ${sectionTitle}` : sectionTitle;
+    }
+
+    return sectionTitle;
+  }, []);
+
+  const scrollToSection = useCallback(debounce((d) => {
+    const sectionIndex = scrollToNextEvent ? getNextSectionIndex(d) : getSectionIndex(d);
+    if (isUndefined(sectionIndex)) {
+      return;
+    }
+
+    if (list?.current && sectionIndex !== undefined) {
+      sectionScroll.current = true; // to avoid setDate() in onViewableItemsChanged
+      _topSection.current = sections[findItemTitleIndex(sectionIndex)]?.title;
+
+      list.current?.scrollToIndex(sectionIndex, true);
+    }
+  }, 1000, {leading: false, trailing: true}), [ sections]);
+
+  const layoutProvider = useMemo(
+    () => new LayoutProvider(
+      (index) => data[index]?.isTitle ? 'title': 'page',
+      (type, dim) => {
+        dim.width = constants.screenWidth;
+        dim.height = type === 'title' ? 60 : 120;
+      }
+    ),
+    [data]
+  );
+
+  const _onScroll = useCallback((rawEvent: any) => {
+    if (!didScroll.current) {
+      didScroll.current = true;
+      scrollToSection.cancel();
+    }
+
+    // Convert to a format similar to NativeSyntheticEvent<NativeScrollEvent>
+    const event = {
+      nativeEvent: {
+        contentOffset: rawEvent.nativeEvent.contentOffset,
+        layoutMeasurement: rawEvent.nativeEvent.layoutMeasurement,
+        contentSize: rawEvent.nativeEvent.contentSize,
+      },
+    };
+    onScroll?.(event as any);
+  }, [onScroll]);
+
+  const _onVisibleIndicesChanged = useCallback(debounce((all: number[]) => {
+    if (all && all.length && !sectionScroll.current) {
+      const topItemIndex = all[0];
+      const topSection = data[findItemTitleIndex(topItemIndex)];
+      if (topSection && topSection !== _topSection.current) {
+        _topSection.current = topSection.title;
+        if (didScroll.current && !avoidDateUpdates) {
+          // to avoid setDate() on first load (while setting the initial context.date value)
+          setDate?.(topSection.title, UpdateSources.LIST_DRAG);
+        }
+      }
+    }
+  }, 1000, {leading: false, trailing: true},), [avoidDateUpdates, setDate, data]);
+
+  const findItemTitleIndex = useCallback((itemIndex: number) => {
+    let titleIndex = itemIndex;
+    while (titleIndex > 0 && !data[titleIndex]?.isTitle) {
+      titleIndex--;
+    }
+
+    return titleIndex;
+  }, [data]);
+
+  const _onMomentumScrollEnd = useCallback(() => {
+    sectionScroll.current = false;
+  }, []);
+
+  const headerTextStyle = useMemo(() => [style.current.sectionText, sectionStyle], [sectionStyle]);
+
+  const onHeaderLayout = useCallback((event: LayoutChangeEvent) => {
+    sectionHeight.current = event.nativeEvent.layout.height;
```

Comment:
Where are you using the sectionHeight.current?

Reply:
removed